core/src/main/java/org/gbif/pipelines/core/ws/client/internal/GBIFInternalServiceClient.java

```diff
@@ -0,0 +1,174 @@
+package org.gbif.pipelines.core.ws.client.internal;
+
+import org.gbif.pipelines.core.ws.HttpClientFactory;
+import org.gbif.pipelines.core.ws.HttpConfigFactory;
+import org.gbif.pipelines.core.ws.config.Config;
+import org.gbif.pipelines.core.ws.config.Service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import okhttp3.OkHttpClient;
+import retrofit2.Call;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * rest client for getting gbif internal api responses
+ */
+public class GBIFInternalServiceClient {
+
+  private static final String INSTALLATION_KEY = "installationKey";
+  private static final String PUB_ORGANIZATION_KEY = "publishingOrganizationKey";
+  private static final String DATASET_TITLE_KEY = "title";
+  private static final String PUB_ORG_COUNTRY_KEY = "country";
+  private static final String INSTALLATION_TYPE_KEY = "type";
+  private static final String DATASET_NETWORK_KEY = "key";
+  private final GBIFInternalService service;
+  //caches the last requests obtained
+  private final Cache<String, GBIFInternalResponse> datasetResponseCache =
+    CacheBuilder.newBuilder().maximumSize(10000).build();
+
+  /**
+   * create client from provided configuration
+   */
+  public static GBIFInternalServiceClient from(Config wsConfig) {
+    // create client
+    OkHttpClient client = HttpClientFactory.createClient(wsConfig);
+
+    // create service
+    Retrofit retrofit = new Retrofit.Builder().client(client)
+      .baseUrl(wsConfig.getBasePath())
+      .addConverterFactory(GsonConverterFactory.create())
+      .validateEagerly(true)
+      .build();
+
+    return new GBIFInternalServiceClient(retrofit.create(GBIFInternalService.class));
+  }
+
+  /**
+   * initialize Client with default values
+   */
+  public static GBIFInternalServiceClient client() {
+    return GBIFInternalServiceClient.from(HttpConfigFactory.createConfig(Service.GBIF_INTERNAL));
+  }
+
+  private GBIFInternalServiceClient(GBIFInternalService internalService) { this.service = internalService;}
+
+  /**
+   * requests https://api.gbif.org/v1/dataset/{datasetid}/networks
+   *
+   * @return array of networks for provided datasetID
+   */
+  public JsonArray getNetworkFromDataset(String datasetUUID) {
```

Comment:
all get* methods must be private, please follow the IDEA warnings

Reply:
I donot see IDEA warnings for this, however i kept it public as they are exposed as main web service client call. This method also helps in testing our main response function getInternalResponse renamed to getDatasetMetaInfo()